cdap-ui/app/directives/epoch/charts.js

```diff
@@ -52,6 +56,9 @@ ngEpoch.controller('epochController', function ($scope, $compile, caskWindowMana
       $scope.$watch('history', function (newVal) {
         if(newVal) {
           $scope.options.data = newVal;
+          if ($scope.options.updateFull) {
```

Comment:
For realtime-line, realtime-area and realtime-bar, we don't render the entire history each poll iteration. We just push into the stream.
Only if updateFull do we render the entire chart each poll (since nothing else makes sense for pie chart).

We can also set updateFull = true for other types of charts if we want though.